Sources/ComposableArchitecture/TestStore.swift

```diff
@@ -1849,9 +1864,19 @@ extension TestStore where ScopedState: Equatable {
     let start = DispatchTime.now().uptimeNanoseconds
     while !Task.isCancelled {
       await Task.detached(priority: .background) { await Task.yield() }.value
-
-      guard self.reducer.receivedActions.isEmpty
-      else { break }
+      
+      switch self.exhaustivity {
+      case .on:
+        guard self.reducer.receivedActions.isEmpty
+        else {
+          return
```

Comment:
I think returning here is okay, because breaking and returning are equivalent since nothing happens after the while loop.